### PR TITLE
plugins: vision: use complete visual odometry msg to send to the FCU

### PR DESCRIPTION
**Pull request by TSC21**
Reboots #201.

### CODE REVIEW

**Comment by TSC21** [comment]
@ChristophTobler this is now working and sending the `VISION_POSITION_ESTIMATE` data properly rotated. The current issue now is related with `ODOMETRY`:  using https://github.com/PX4/Firmware/pull/9301 I don't seem to get the msgs on the autopilot side. Can you have a test on your side by setting https://github.com/PX4/sitl_gazebo/pull/203/files#diff-5ed56584b86a34e30277b2fd7cb7a917R129 to true, and check if you get anything on the autopilot side? Thanks!

**Comment by ChristophTobler** [comment]
@TSC21 This is still not working... There's a bug in the transformation

With
```
// transform position from local ENU to local NED frame
vision.x = odom_message->position().y();
vision.y = -odom_message->position().x();
vision.z = -odom_message->position().z();

// q_nb is the quaternion that represents a rotation from NED earth/local
// frame to XYZ body FRD frame
ignition::math::Vector3d euler = q_nb.Euler();

vision.roll = -euler.X();
vision.pitch = euler.Y();
vision.yaw = euler.Z();
```
it worked

**Comment by TSC21** [comment]
The rotation quaternions are right so I don't get why is this wrong. How are you debugging this?

File: src/gazebo_mavlink_interface.cpp [hunk 1]
```diff
@@ -767,20 +743,120 @@ void GazeboMavlinkInterface::IRLockCallback(IRLockPtr& irlock_message) {
 }
 
 void GazeboMavlinkInterface::VisionCallback(OdomPtr& odom_message) {
-  mavlink_vision_position_estimate_t sensor_msg;
-  sensor_msg.usec = odom_message->usec();
-  // convert from ENU to NED
-  sensor_msg.x = odom_message->y();
-  sensor_msg.y = -odom_message->x();
```
Comment (TSC21) [inline src/gazebo_mavlink_interface.cpp:774]: This is what is causing me confusion: from the local ENU to local NED, you swap X with Y and negate Z. For the above to happen, it means you have the position on the FLU body frame, which means you first have to transform to body FRD (180º roll rotation) and then apply the transform from ENU to NED.

Comment (TSC21) [reply src/gazebo_mavlink_interface.cpp:774]: Still, something is not right here, cause if we apply the 180º rotation on Roll, it negates Y and Z. So this transform is not making sense to me, even though it is working.

**Comment by TSC21** [comment]
@ChristophTobler I actually removed the transform from local to body frame and this now works as it should. The position should be propagated on the local ENU frame while the orientation should represent the rotation from the ENU frame to the body FLU, and then transformed in the rotation from the NED frame to the body FRD. The rotations are now the expected.

**Comment by TSC21** [comment]
@ChristophTobler are you able to give a try but with `send_odometry_` set to `true`? I am struggling to understand why the `ODOMETRY` msgs are not getting to the FCU. A 2nd tester is welcomed cause I am probably missing something simple.

**Comment by TSC21** [comment]
Problem solved by updating the protocol version on the `gazebo_mavlink_interface` side. Also added an SDF parameter for it in case one wants to downgrade the protocol version to 1. By default, I set it to 2.

**Comment by TSC21** [comment]
Here's a log of an hover test with `iris_opt_flow`, having Gazebo publishing `ODOMETRY` msgs: https://logs.px4.io/plot_app?log=8512ebe5-02a1-4db5-a9d6-1db2259c60a8. This was tested with the branch from https://github.com/PX4/Firmware/pull/9301.

**Review by LorenzMeier — APPROVED** [review]

**Comment by TSC21** [comment]
@LorenzMeier thanks!

**Pull request merged**